File: ruby/majority_element.rb
nums = [2, 2, 1, 1, 1, 2, 2]

def majority_element(nums)
  hash = {}
  nums.each do |num|
    if hash[num]
      hash[num] += 1
    else
      hash[num] = 0
    end
  end

  hash.key(hash.values.max)
end

p majority_element(nums)

File: python/majority_element.py
nums = [2, 2, 1, 1, 1, 2, 2]

def majority_element(nums):
  dic = {}
  for num in nums:
    if num in dic:
      dic[num] += 1
    else:
      dic[num] = 0

  return max(dic, key = dic.get)

print(majority_element(nums))